setup/building-stratum-deps.md

```diff
@@ -0,0 +1,183 @@
+# Building the Stratum Dependencies
+
+Stratum is the component of `infrap4d` that implements the P4Runtime and gNMI
+(OpenConfig) services. It requires that a number of third-party libraries
+be built and installed on the system.
+
+This document explains how to build and install the Stratum dependencies.
+
+> Note: For the Intel IPU E2100, see
+[Building Stratum Dependencies for ES2K](building-stratum-deps-es2k.md).
```

Comment:
Self reference?

Reply:
On that note, I think it makes more sense to have the document in `docs` folder rather than in `setup`. If looking for any build instructions, users will generally tend to look in docs

Reply:
> Self reference?

No, it's a reference to an ES2K-specific document that I'm working on.

I mentioned this in the comment at the top of this page.

> Generic instructions for building the Stratum dependencies using CMake.
ES2K-specific instructions (using the helper scripts) will follow.

Reply:
> On that note, I think it makes more sense to have the document in `docs` folder rather than in `setup`. If looking for any build instructions, users will generally tend to look in docs

I feel very strongly that the instructions for building the dependencies belong with the dependencies. Building and installing the dependencies is a precursor to building and installing P4 Control Plane, and is no more a part of P4 Control Plane than the SDE is.

In fact, I'm proposing to spin the Stratum Dependencies off as an independent package with its own repository, and refer to it rather than including it in the networking recipe.

See https://wiki.ith.intel.com/display/ndvswitch/P4CP+-+Stratum+Dependencies for more information.

Reply:
hmmm I see. I would atleast like to see a direct link from the "main build" document in `docs` folder to this then.
We can revisit once all documentation is in place. Approving this PR.

Reply:
The plan is to turn `es2k-guide.md` into a list of steps to take, with each step linking to the document for that step.